src/app/modules/assignments/reducer.ts

```diff
@@ -101,6 +101,23 @@ const reducer = createReducer<AssignmentState>({
     let newTemplates = templates.slice();
     newTemplates.push(payload);
     return { templates: newTemplates, ...rest, saving:false, successMessage:'SAVED!' };
+  },
+  ASSIGNMENT_TEMPLATE_EDIT_BEGIN: (state, payload) => (Object.assign({}, state, {saving: true })),
+  ASSIGNMENT_TEMPLATE_EDIT_FAIL: (state, payload) => ({ saving:false, error:payload }),
+  ASSIGNMENT_TEMPLATE_EDIT_SUCCESS: (state, payload) => {
+    const { templates=[], ...rest } = state;
+    templates[payload.id] = payload;
```

Comment:
Here instead of mutating our state i.e. `templates`, mutate the copy of templates i.e.
```ts
let newTemplates = templates.slice();
newTemplates[payload.id] = payload;
....
```